src/vault/authenticate_vault/token.rs

```diff
@@ -0,0 +1,41 @@
+use crate::errors::error::VaultError;
+use crate::vault::vault_config::VaultConfig;
+use crate::vault::vault_service::VaultService;
+use vaultrs::{
+    client::{Client, VaultClient, VaultClientSettingsBuilder},
+};
+use async_trait::async_trait;
+use crate::vault::authenticate_vault::authenticate_vault_trait::AuthenticateVault;
+
+pub struct AuthenticateTokenVault;
+
+#[async_trait]
+impl AuthenticateVault for AuthenticateTokenVault {
+    fn get_vault_settings(&self, config: VaultConfig) -> VaultClientSettingsBuilder {
+        return VaultClientSettingsBuilder::default()
+            .address(config.address.clone())
+            .token(config.token.clone().unwrap())
+            .timeout(Some(config.client_timeout)).clone()
+    }
+
+    fn get_jwt_token(&self, _: VaultConfig) -> Result<Option<String>, VaultError> {
+        return Ok(None);
```

Comment:
Why Ok is returned here? I'm trying to understand difference between this and get_jwt_token method in authenticate_vault_trait.rs.

Reply:
hence this trait is implemented for root token authentication, jwt does not necessary and return as None, as far as i understand 😄